### PR TITLE
Add env vars

### PR DESCRIPTION
**Pull request by s1037989**
This PR addresses most of the topics I posted in IRC today.
- I added two small features and what I consider a fix but may not be done the correct way.
- I improved the docs, but unfortunately I included the docs for the features in with the doc fixes.

I'll be happy to fix this up however you like, but thought I'd run it by you first to get your overall thoughts.

Thanks!

### CODE REVIEW

File: lib/App/git/ship/perl.pm [hunk 1]
```diff
@@ -139,7 +139,7 @@ sub ship {
   $self->system(qw( git add Makefile.PL ), $changelog, $self->_filename('readme'));
   $self->system(qw( git commit -a -m ), $self->_changes_to_commit_message);
   $self->SUPER::ship(@_);    # after all the changes
-  $uploader->upload_file($dist_file);
+  $uploader->upload_file($dist_file) unless $ENV{GIT_SHIP_NO_UPLOAD};
```
Comment (jhthorsen) [inline lib/App/git/ship/perl.pm]: How about `unless $ENV{GIT_SHIP_UPLOAD} // 1` ?

Comment (jhthorsen) [inline lib/App/git/ship/perl.pm]: Maybe we should turn $uploader into an attribute and then have:

```
$self->uploader->upload_file($dist_file) unless $ENV{GIT_SHIP_UPLOAD} // 1;
```

**Comment by jhthorsen** [comment]
Looks good to me in general. Just reply to the comments and you can merge :+1:

File: lib/App/git/ship/perl.pm (hunk 1 above)
Comment (s1037989) [inline lib/App/git/ship/perl.pm]: turn $uploader into an attribute
- :+1:

unless $ENV{GIT_SHIP_UPLOAD} // 1;
- `do the upload unless UPLOAD is set`, right?  So if I set UPLOAD, it won't upload?  If my brain logic is functioning properly, that seems counter-intuitive.  I was thinking `do the upload unless NO_UPLOAD is set`, or, said another way, `if NO_UPLOAD is set, don't upload`.

Comment (jhthorsen) [inline lib/App/git/ship/perl.pm]: The logic will be if GIT_SHIP_UPLOAD=0 then don't upload, but it defaults to 1 unless defined. Anyway, I don't mind having GIT_SHIP_NO_UPLOAD. Maybe it should be something you can set in .ship.conf as well?

**Pull request merged**